src/utils.js

```diff
@@ -66,3 +66,65 @@ export const objectQueryString = obj => {
     .map(([k, v]) => `${k}:${serializeAttribute(v)}`)
     .join(';');
 };
+
+const strSplitOnFirst = (s, char) => {
+  const i = s.indexOf(char);
+  if (i === -1) {
+    return [s];
+  }
+
+  return [s.substring(0, i), s.substring(i + 1)];
+};
+
+/**
+   Compute longest common path prefix for an array of path strings.
+   The input array must be sorted lexically.
+*/
+const longestPathPrefix = sortedPaths => {
+  let first = sortedPaths[0];
+  let last = sortedPaths[sortedPaths.length - 1];
+  let result = '';
+
+  if (sortedPaths.length === 0) {
+    return result;
+  }
+  if (sortedPaths.length === 1) {
+    const comps = sortedPaths[0].split('/');
+    if (comps.length === 1) {
+      return result;
+    }
+    const path = comps.slice(0, comps.length - 1).join('/');
+    return `${path}/`;
+  }
+
+  while (first && last) {
+    const [firstCurrent, firstRest] = strSplitOnFirst(first, '/');
+    const [lastCurrent, lastRest] = strSplitOnFirst(last, '/');
+    if (firstCurrent !== lastCurrent) {
+      break;
+    }
+    first = firstRest;
+    last = lastRest;
+    result = `${result}${firstCurrent}/`;
```

Comment:
Comment: Quite a lot of mutation is happening here :) Probably the same could've been achieved with recursion or reduce. Well, it looks good and simple enough so I'm not suggesting a change here.

Reply:
Good point. We can make it recursive, that's perhaps cleaner. With reduce is doable, but probably more complex than it is now.